backend/workflow_manager/endpoint_v2/source.py

```diff
@@ -417,21 +433,28 @@ def get_file_content(self, input_file_path: str, chunk_size: int = 8192) -> byte
             (default is 8192 bytes).
 
         Returns:
-            bytes: The content of the file.
+            tuple[bytes, int]:
+                A tuple containing the content of the file as bytes and
+                its size in bytes.
         """
         connector: ConnectorInstance = self.endpoint.connector_instance
         connector_settings: dict[str, Any] = connector.connector_metadata
         source_fs = self.get_fsspec(
             settings=connector_settings, connector_id=connector.connector_id
         )
+
+        # Get file size
+        file_metadata = source_fs.stat(input_file_path)
+        file_size = file_metadata.get("size", 0)
```

Comment:
```suggestion
        file_size = file_metadata.get("size")
```
@muhammad-ali-e since we store an `Optional[int]` in `FileHash` it might be good to leave it unassigned if we're not able to obtain the size for a file - please change the return type as well accordingly